### PR TITLE
feat: use new ipfsd-ctl

### PR DESCRIPTION
**Pull request by dryajov**

### CODE REVIEW

File: test/bitswap.spec.js [hunk 1]
```diff
@@ -5,30 +5,28 @@ const chai = require('chai')
...
+  let ipfsd = null
 
   before((done) => {
     this.timeout(20 * 1000) // slow CI
-    fc = new FactoryClient()
-    fc.spawnNode((err, node) => {
+    df.spawn((err, node) => {
       expect(err).to.not.exist()
-      ipfs = node
+      ipfsd = node
```
Comment (diasdavid) [inline test/bitswap.spec.js]: replace this line by: `ipfs = ipfsd.api` and avoid replacing all the other lines.

**Comment by diasdavid** [comment]
I've pulled the latest and did all the linking and I get an error:

```
  1) .swarm
       callback API
         .addrs:
     Uncaught Error: no protocol with name: dnsaddr
      at Protocols (/Users/imp/code/js-ipfsd-ctl/node_modules/multiaddr/src/protocols-table.js:17:11)
      at stringToStringTuples (/Users/imp/code/js-ipfsd-ctl/node_modules/multiaddr/src/codec.js:45:19)
      at stringToBuffer (/Users/imp/code/js-ipfsd-ctl/node_modules/multiaddr/src/codec.js:170:13)
      at Object.fromString (/Users/imp/code/js-ipfsd-ctl/node_modules/multiaddr/src/codec.js:178:10)
      at new Multiaddr (/Users/imp/code/js-ipfsd-ctl/node_modules/multiaddr/src/index.js:39:25)
      at Multiaddr (/Users/imp/code/js-ipfsd-ctl/node_modules/multiaddr/src/index.js:27:12)
      at result.Addrs.(anonymous function).forEach (/Users/imp/code/js-ipfsd-ctl/node_modules/ipfs-api/src/swarm/addrs.js:25:35)
      at Array.forEach (<anonymous>)
      at Object.keys.map (/Users/imp/code/js-ipfsd-ctl/node_modules/ipfs-api/src/swarm/addrs.js:24:26)
      at Array.map (<anonymous>)
      at send (/Users/imp/code/js-ipfsd-ctl/node_modules/ipfs-api/src/swarm/addrs.js:22:47)
      at f (/Users/imp/code/js-ipfsd-ctl/node_modules/once/once.js:25:25)
      at streamToValue (/Users/imp/code/js-ipfsd-ctl/node_modules/ipfs-api/src/utils/stream-to-json-value.js:30:5)
      at concat (/Users/imp/code/js-ipfsd-ctl/node_modules/ipfs-api/src/utils/stream-to-value.js:12:22)
      at ConcatStream.<anonymous> (/Users/imp/code/js-ipfsd-ctl/node_modules/concat-stream/index.js:36:43)
      at finishMaybe (/Users/imp/code/js-ipfsd-ctl/node_modules/readable-stream/lib/_stream_writable.js:607:14)
      at endWritable (/Users/imp/code/js-ipfsd-ctl/node_modules/readable-stream/lib/_stream_writable.js:615:3)
      at ConcatStream.Writable.end (/Users/imp/code/js-ipfsd-ctl/node_modules/readable-stream/lib/_stream_writable.js:571:41)
      at IncomingMessage.onend (_stream_readable.js:595:10)
      at endReadableNT (_stream_readable.js:1056:12)
      at _combinedTickCallback (internal/process/next_tick.js:138:11)
      at process._tickDomainCallback (internal/process/next_tick.js:218:9)


```

**Comment by dryajov** [comment]
@diasdavid based on my convo with lgierth on IRC, it seems like `/dns/<addr>` was renamed to `/dnsaddr/<addr>`, here is a PR that addresses it - https://github.com/multiformats/js-multiaddr/pull/50.

**Comment by diasdavid** [comment]
ipfsd-ctl 0.27 has been published. Could you update the PR and rebase master to resolve merge conflicts?

**Comment by dryajov** [comment]
depends on https://github.com/ipfs/interface-ipfs-core/pull/186

**Comment by diasdavid** [comment]
image


@dryajov can you check why the Coveralls dropped so much?

**Comment by dryajov** [comment]
@diasdavid not sure why coverage dropped so much, I see that some tests are failing which might be messing  up the coverage report, but other than that the actual tests haven't been altered to cause such a drop. I'll keep investigating.

**Comment by diasdavid** [comment]
closing this PR and opening again

**Pull request closed without merging**